### PR TITLE
Add Pylint

### PR DESCRIPTION
**Pull request by iquintero**
This adds pylint to the build process - will fail for any Warning/Error on the main codebase. Tests are no included yet. 

The handling of the MXNet distribution parameter was being done in Framework instead of MXNet. Pylint flagged this so this PR also fixes that. distributions is now also called distribution which is what was originally intended.

*Issue #, if available:*

*Description of changes:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    no-self-use,
+    attribute-defined-outside-init,
+    protected-access,
+    invalid-all-object,
+    arguments-differ,
+    abstract-method,
+    signature-differs
+
+
+
```
Comment (laurenyu) [inline .pylintrc]: is the inconsistent number of newlines between sections intentional?

File: .pylintrc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+# Add a comment according to your evaluation note. This is used by the global
+# evaluation report (RP0004).
+#comment=no
+
+[MISCELLANEOUS]
+# List of note tags to take in consideration, separated by a comma.
+#notes=FIXME,XXX,TODO
+[FORMAT]
+# Maximum number of characters on a single line.
+max-line-length=100
```
Comment (laurenyu) [inline .pylintrc]: does this need to be set here if we're letting flake8 handle line length?

File: .pylintrc [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+# Maximum number of lines in a module
+#max-module-lines=1000
+# String used as indentation unit. This is usually " " (4 spaces) or "\t" (1
+# tab).
+indent-string='  '
+
+[TYPECHECK]
+# Tells whether missing members accessed in mixin class should be ignored. A
+# mixin class is detected if its name ends with "mixin" (case insensitive).
+#ignore-mixin-members=yes
```
Comment (laurenyu) [inline .pylintrc]: is it worth keeping around these commented-out config lines?

File: .pylintrc [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    quit,
+# Regular expression which should only match correct module names
+#module-rgx=(([a-z_][a-z0-9_]*)|([A-Z][a-zA-Z0-9]+))$
+# Regular expression which should only match correct module level names
+#const-rgx=(([A-Z_][A-Z0-9_]*)|(__.*__))$
+# Regular expression which should only match correct class names
+#class-rgx=[A-Z_][a-zA-Z0-9]+$
+# Regular expression which should only match correct function names
+#
+#
```
Comment (laurenyu) [inline .pylintrc]: do we need these empty comment lines?

File: .pylintrc [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+# not be disabled)
+#ext-import-graph=
+# Create a graph of internal dependencies in the given file (report RP0402 must
+# not be disabled)
+#int-import-graph=
+
+[LOGGING]
+# Apply logging string format checks to calls on these modules.
+logging-modules=
+    logging
```
Comment (laurenyu) [inline .pylintrc]: add a newline to the end of the file

File: CHANGELOG.rst [hunk 6]
```diff
@@ -2,6 +2,10 @@
 CHANGELOG
 =========
 
+1.14.1dev
+=========
+* bug-fix: rename MXNet argument from accelerators -> accelerator
```
Comment (laurenyu) [inline CHANGELOG.rst]: * add a newline between ll. 6-7
* also don't you mean `distributions` -> `distribution`?
* does this count as a breaking change?

File: src/sagemaker/estimator.py [hunk 7]
```diff
@@ -425,7 +410,7 @@ def _ensure_latest_training_job(self, error_message='Estimator is not associated
 
 
 class _TrainingJob(_Job):
-    def __init__(self, sagemaker_session, training_job_name):
+    def __init__(self, sagemaker_session, training_job_name):  # pylint: disable=useless-super-delegation
```
Comment (laurenyu) [inline src/sagemaker/estimator.py]: why is this one important?

File: src/sagemaker/tuner.py [hunk 8]
```diff
@@ -73,22 +73,13 @@ class ContinuousParameter(_ParameterRange):
     """
     __name__ = 'Continuous'
 
-    def __init__(self, min_value, max_value):
-        """Initialize a ``ContinuousParameter``.
-
-        Args:
-            min_value (float): The minimum value for the range.
-            max_value (float): The maximum value for the range.
-        """
```
Comment (laurenyu) [inline src/sagemaker/tuner.py]: would it be useful to still have the types in the docstrings noted somewhere? the base class says int or float but both subclasses here require one or the other and not both.

File: .pylintrc (hunk 2 above)
Comment (iquintero) [reply .pylintrc]: Not really, but I think its better to just set it to what we care instead of 80.

File: CHANGELOG.rst (hunk 6 above)
Comment (iquintero) [reply CHANGELOG.rst]: Yes and yes

File: .pylintrc (hunk 2 above)
Comment (laurenyu) [reply .pylintrc]: okay. I suppose we won't be changing this frequently, if ever.

File: .pylintrc [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+# CHANGE: No report.
+reports=no
+
+[FORMAT]
+# Maximum number of characters on a single line.
+max-line-length=100
+# Maximum number of lines in a module
+#max-module-lines=1000
+# String used as indentation unit. This is usually " " (4 spaces) or "\t" (1
+# tab).
```
Comment (laurenyu) [inline .pylintrc]: do we really need to break the line on this comment?

**Pull request merged**